src/components/Scheme/ContributionRewardExtRewarders/Competition/Card.tsx

```diff
@@ -0,0 +1,139 @@
+import AccountPopup from "components/Account/AccountPopup";
+import AccountProfileName from "components/Account/AccountProfileName";
+import { humanProposalTitle } from "lib/util";
+import * as React from "react";
+import { Link } from "react-router-dom";
+import RewardsString from "components/Proposal/RewardsString";
+import { IDAOState, IProposalState, CompetitionVote, ICompetitionSuggestionState } from "@daostack/client";
+import { IProfileState } from "reducers/profilesReducer";
+import { IRootState } from "reducers";
+import { connect } from "react-redux";
+import withSubscription, { ISubscriptionProps } from "components/Shared/withSubscription";
+import { combineLatest } from "rxjs";
+import CountdownText from "components/Scheme/ContributionRewardExtRewarders/Competition/CountdownText";
+import StatusBlob from "./StatusBlob";
+import * as css from "./Competitions.scss";
+import { competitionStatus, getProposalSubmissions, getCompetitionVotes, CompetitionStatus } from "./utils";
+
+type ISubscriptionState = [Array<ICompetitionSuggestionState>, Array<CompetitionVote>];
+
+interface IExternalStateProps {
+  creatorProfile: IProfileState;
+}
+
+interface IStateProps {
+  status: CompetitionStatus;
+}
+
+interface IExternalProps {
+  daoState: IDAOState;
+  proposalState: IProposalState;
+  handleStatusChange: (proposal: IProposalState, newStatus: CompetitionStatus) => void;
+}
+
+type IProps = IExternalProps & IExternalStateProps & ISubscriptionProps<ISubscriptionState>;
+
+const mapStateToProps = (state: IRootState, ownProps: IExternalProps): IExternalProps & IExternalStateProps => {
+
+  return {
+    ...ownProps,
+    creatorProfile: state.profiles[ownProps.proposalState.proposer],
+  };
+};
+
+class CompetitionCard extends React.Component<IProps, IStateProps> {
+
+  constructor(props: IProps) {
+    super(props);
+    const status = this.getCompetitionState();
+    this.state = { status };
+    // give the status to who wants it
+    this.fireHandleChange(status);
+  }
+
+  private getCompetitionState = (): CompetitionStatus => {
+    const competition = this.props.proposalState.competition;
+    const submissions = this.props.data[0];
+    return competitionStatus(competition, submissions);
+  }
+
+  private fireHandleChange(status: CompetitionStatus) {
+    this.props.handleStatusChange(this.props.proposalState, status);
+  }
+
+  private onEndCountdown = () => {
+    // paranoia about timer inprecision
```

Comment:
It is why I'm using setTimer before recomputing the competition state which depends on the current browser/javascript time.  I have worried that the timers may be inprecise in arriving at zero at precisely when they are supposed to.  I think this may be paranoia.

Reply:
it _is_ indeed paranoid, probably; the ethereum chanin has only a  vague concept of time (one tick every 15 seconds, about, give or take a second), so our times are much more precise (and therefore always wrong, strictly speaking)

Reply:
Before I added the setTimeout I was seeing the UI not always updating on this event.  That stopped after I added this.  Maybe a coincidence.